helmctlr/helm.go

```diff
@@ -89,10 +90,13 @@ func (c Controller) ResourceAdded(r *unstructured.Unstructured) {
 	if err := c.installOrUpdate(r); err != nil {
 		metrics.CreateFailures.Inc()
 		c.logger.Errorw("failed to create resource", "error", err, "resource", r.GetName())
+		c.updateCRStatus(r, crw.PhaseFailed, crw.ReasonApplyFailed, err.Error())
```

Comment:
can't these statuses be updated in `installOrUpgrade` instead of both `ResourceAdded` and `ResourceUpdated` to reduce the duplication?

Reply:
I'd prefer to keep the calls at the top level to make it clear what is happening.